Sources/Algorithms/Product.swift

```diff
@@ -91,27 +91,22 @@ extension Product2: Collection where Base1: Collection {
   }
   
   public var startIndex: Index {
-    base1.isEmpty || base2.isEmpty
-      ? endIndex
-      : Index(i1: base1.startIndex, i2: base2.startIndex)
+    Index(
+      i1: base2.isEmpty ? base1.endIndex : base1.startIndex,
+      i2: base2.startIndex)
   }
   
   public var endIndex: Index {
-    Index(i1: base1.endIndex, i2: base2.endIndex)
+    // this representation makes index calculations simpler
```

Comment:
I'm not sure we need this comment to make it into the final patch

Reply:
Thanks for the feedback, the only reason for putting it there was to indicate to someone reading the code that this is not a typo – any idea how else we could convey this?

Reply:
I'll defer to @natecook1000 on whether we should keep it, I guess I generally don't assume there are typos in the code, and if I'm surprised I'll read the surrounding code closer.

That being said, I can understand wanting to draw attention to i2 being base.startIndex, to prevent the reader from glossing over it. For example:

```
// `base2.startIndex` simplifies index calculations.
```

Reply:
Thanks, that's indeed what I was going for. I'll change it to that for now and then leave it up to Nate whether to remove it completely.